### PR TITLE
Next song fixes #32

### LINKED ISSUE #32: Switch songs automatically
Change the frontend so it can play a new song when the old song has finished.

### PR DESCRIPTION
**Pull request by Azkle**
It works, but the backend doesn't give a new song.

Tried to fix this in branch next-song, but no succes so far.

### CODE REVIEW

**Comment by Azkle** [comment]
The backend has been fixed by @JeremybellEU in https://github.com/MoodCat/MoodCat.me-Core/pull/132

File: app/scripts/controllers/main.js [hunk 1]
```diff
@@ -39,8 +39,22 @@ angular.module('moodCatApp')
...
+          room.nextSong = room.song.soundCloudId;
+          this.callNextSong(room);
         return room;
-      }
+      }   
+
+      this.callNextSong = function callNextSong(room){
+
+        if((room.song.duration/1000 - currentSongService.getTime()) < 1){
+          $http.get('/api/rooms/' + room.id).then(function(response) {
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: Just reuse the existing fetchRoom method here, this is now duplicate code.

File: app/mocks/song.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"id":119,"song":{"id":624,"soundCloudId":69238898,"artist":{"id":624,"name":"D33J"},"name":"Love Story (Syrup Cuz Im Single D33J Remix)","duration":231543,"artworkUrl":"https://i1.sndcdn.com/artworks-000035119901-r7lf9j-large.jpg"},"name":"ROOM_STUB #8","time":121}
```
Comment (JWGmeligMeyling) [inline app/mocks/song.json]: This file seems to be unused now.

File: app/scripts/controllers/moodCatAudio.js [hunk 3]
```diff
@@ -88,6 +88,16 @@
        return pad(hours, 2) + ':' + pad(minutes, 2) + ':' + pad(seconds, 2);// + '.' + pad(millis, 3);
      }
 
+     /**
+      * Get the time
+      * @returns {integer} The time of the song.
+      */
+    this.getTime = function getTime() {
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/moodCatAudio.js]: It seems that this function is unused, remove it?

File: app/scripts/controllers/main.js [hunk 4]
```diff
@@ -39,8 +39,22 @@ angular.module('moodCatApp')
...
         }
+          room.nextSong = room.song.soundCloudId;
+          this.callNextSong(room);
         return room;
-      }
+      }   
+
+      this.callNextSong = function callNextSong(room){
+
+        if((room.song.duration/1000 - currentSongService.getTime()) < 1){
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: This should be refactored, it's very hard to read.

File: app/scripts/controllers/main.js [hunk 5]
```diff
@@ -39,8 +39,22 @@ angular.module('moodCatApp')
...
+          $http.get('/api/rooms/' + room.id).then(function(response) {
+             room.nextSong = response.data.song.soundCloudId;
+             currentSongService.loadSong(room.nextSong);
+             room.song = room.nextSong;
+           });
+        }
+        setTimeout(callNextSong, 1000, room);
```
Comment (JWGmeligMeyling) [inline app/scripts/controllers/main.js]: This schedule will break when switching rooms, because the callNextSong for the old room will still be scheduled.

**Comment by Azkle** [comment]
We should merge master into this branch, so that the build doesn't fail.

**Comment by JWGmeligMeyling** [comment]
Merge good to go!

**Comment by GijsWeterings** [comment]
LGTM as well

**Pull request merged**